src/MongoDB.Driver.Core/Core/Clusters/ClusterFactory.cs

```diff
@@ -53,14 +53,7 @@ public ICluster CreateCluster()
             switch (connectionMode)
             {
                 case ClusterConnectionMode.Automatic:
-                    if (settings.EndPoints.Count == 1 && settings.Scheme != ConnectionStringScheme.MongoDBPlusSrv)
-                    {
-                        return CreateSingleServerCluster(settings);
-                    }
-                    else
-                    {
-                        return CreateMultiServerCluster(settings);
-                    }
+                    return CreateMultiServerCluster(settings);
```

Comment:
This is a behavioral change that I think should be highlight with its own Jira issue, and perhaps a mention in the release notes.

What do you think @rstam ?

Reply:
I'm not sure it's anything the user would even notice.

Reply:
Let me state clearly what I think the effect of this change is:

Constructing a new MongoClient with all default settings creates a cluster with `Automatic ConnectionMode connected to localhost:27017.  Let's say localhost:27017 is a member of a replica set.  Currently the driver will make a direct connection to that server. With this change, the driver will do discovery and connect to all replica set members.

If so, this is exactly the behavioral change that the specification says should be done in a major release (i.e. change the default value of directConnection from true to false).

Please let me know if I'm misunderstanding.

Reply:
I see your point. I guess I didn't think it was a problem because that was how I thought the driver worked all along. So to me it looked like a bug fix rather than a behavioral change.

Let's discuss in more detail tomorrow.